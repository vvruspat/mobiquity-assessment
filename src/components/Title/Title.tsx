import { FC, PropsWithChildren } from "react";
import { Text } from "../Text";

import s from "./Title.module.css";

type TitleProps = {};

/**
 * Text component
 */
export const Title: FC<PropsWithChildren<TitleProps>> = ({ children }) => {
  return (
    <h2 className={s.title}>
      <Text size="xl" weight="bold">
        {children}
      </Text>
    </h2>
  );
};
