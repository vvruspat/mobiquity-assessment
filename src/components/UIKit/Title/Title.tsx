import {
  DetailedHTMLProps,
  FC,
  HTMLAttributes,
  PropsWithChildren,
} from "react";
import cn from "clsx";

import { Text } from "../Text";

import s from "./Title.module.css";

type TitleProps = DetailedHTMLProps<
  HTMLAttributes<HTMLHeadingElement>,
  HTMLHeadingElement
>;

/**
 * Text component
 */
export const Title: FC<PropsWithChildren<TitleProps>> = ({
  children,
  className,
  ...htmlProps
}) => {
  return (
    <h2 className={cn(s.title, className)} {...htmlProps} data-testid="title">
      <Text size="xl" weight="bold">
        {children}
      </Text>
    </h2>
  );
};
