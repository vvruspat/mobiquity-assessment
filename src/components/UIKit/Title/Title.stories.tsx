import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Title } from "./Title";
import "../../../index.css";

export default {
  title: "UI Kit/Typography/Title",
  component: Title,
} as ComponentMeta<typeof Title>;

const Template: ComponentStory<typeof Title> = (args) => (
  <Title {...args}>Title</Title>
);

export const TitleTemplate = Template.bind({});

TitleTemplate.args = {};
