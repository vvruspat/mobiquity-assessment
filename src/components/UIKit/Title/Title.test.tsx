import { render, screen } from "@testing-library/react";
import { Title } from "./Title";

test("renders title", () => {
  render(<Title>Test title</Title>);

  const el = screen.getByTestId("title");

  expect(el).toHaveClass("title");
  expect(el).toHaveTextContent("Test title");
  expect(el).toMatchSnapshot();
});
