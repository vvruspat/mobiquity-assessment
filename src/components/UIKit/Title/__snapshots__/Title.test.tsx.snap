// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders title 1`] = `
<h2
  class="title"
  data-testid="title"
>
  <span
    class="text size_xl bold"
    data-testid="text"
  >
    Test title
  </span>
</h2>
`;
